# apply this patch if you are geting this error:
# /src/pkg/exp/4s//xs.go:729: cannot use int32((time.Nanoseconds()) % 999999999) (type int32) as type int64 in function argument
#
diff -r 2f32e74ab96e src/pkg/exp/4s/xs.go
--- a/src/pkg/exp/4s/xs.go	Sat Nov 21 15:53:03 2009 -0800
+++ b/src/pkg/exp/4s/xs.go	Mon Nov 23 17:39:28 2009 +0100
@@ -726,7 +726,7 @@
 	pieces = pp;
 	N = len(pieces[0].d);
 	initPieces();
-	rand.Seed(int32(time.Nanoseconds() % (1e9 - 1)));
+	rand.Seed(int64(time.Nanoseconds() % (1e9 - 1)));
 	whitemask = draw.White.SetAlpha(0x7F);
 	tsleep = 50;
 	timerc = time.Tick(int64(tsleep/2) * 1e6);
